src/metabase/lib/js.cljs

```diff
@@ -1210,6 +1210,14 @@
   [arg]
   (and (map? arg) (= :metadata/column (:lib/type arg))))
 
+(defn ^:export is-metric-metadata
+  "Returns true if arg is an MLv2 metric, ie. has `:lib/type :metadata/metric`.
+
+  > **Code health:** Smelly. When is this called and why does the FE need to know? The values are supposed to be opaque,
+  and we should see if there's a better way to get the needed information."
+  [arg]
+  (and (map? arg) (= :metadata/metric (:lib/type arg))))
```

Comment:
I think the correct check is based on `:type` and the value should be `:metric` https://github.com/metabase/metabase/blob/f617c871735914be8d812f06d6dc6648dc16ec40/src/metabase/lib/card.cljc#L178

Reply:
I'm honestly a bit out of my depths here. I based this off of the following:

https://github.com/metabase/metabase/blob/0c9bc4fab2fd1052595c6247e80816e7855e72bf/src/metabase/lib/js/metadata.cljs#L461-L465

The tests seem to pass here but they might be wrong too.